app/models/ocelot/package.scala

```diff
@@ -21,9 +21,13 @@ import scala.util.matching.Regex
 package object ocelot {
   val hintRegex = "\\[hint:([^\\]])+\\]".r
   val pageLinkRegex = s"\\[link:.+?:(\\d+|${Process.StartStanzaId})\\]".r
-  val labelRefRegex = s"\\[label:([0-9a-zA-Z\\s+]+)\\]".r
+  val labelRefRegex = s"\\[label:([0-9a-zA-Z\\s+_]+)\\]".r
+  val currencyRegex = "\\d+(\\.(\\d{2})?)?".r // From Ocelot
+
 
   def plSingleGroupCaptures(regex: Regex, str: String): List[String] = regex.findAllMatchIn(str).map(_.group(1)).toList
   def pageLinkIds(str: String): List[String] = plSingleGroupCaptures(pageLinkRegex, str)
   def labelReferences(str: String): List[String] = plSingleGroupCaptures(labelRefRegex, str)
+  def labelReference(str: String): Option[String] = plSingleGroupCaptures(labelRefRegex, str).headOption
+  def isCurrency(str: String): Boolean = currencyRegex.findFirstIn(str).fold(false)(_=> true)
```

Comment:
This is an interesting case. Running the function in REPL the value 10.000000 is accepted as currency.

Reply:
yes the Ocelot pattern allows many digits after the dp, could limit it, but kept as Ocelot